src/gui/activitylistmodel.cpp

```diff
@@ -291,18 +291,28 @@ void ActivityListModel::combineActivityLists()
     endInsertRows();
 }
 
+bool ActivityListModel::canFetchActivities() const {
+    return _accountState->isConnected() && _accountState->account()->capabilities().hasActivities();
+}
+
 void ActivityListModel::fetchMore(const QModelIndex &)
 {
-    if (_accountState->isConnected()) {
-        _activityLists = ActivityList();
+    _activityLists = ActivityList();
```

Comment:
i'm quite confused why fetchMore() clears previously fetched activities

Reply:
That is actually a good question